stats/bool.go

```diff
@@ -0,0 +1,36 @@
+package stats
+
+import (
+	"sync/atomic"
+	"time"
+)
+
+type Bool struct {
+	val uint32
```

Comment:
because there is no support to atomatically change a type like that. see https://golang.org/pkg/sync/atomic/